### DISCUSSION TITLE
Received message handler?

### DISCUSSION THREAD

**Discussion by Diden05** [dimaskiddo/go-whatsapp-multidevice-rest#58] [Ideas]
We need a handler for received messages, calls, message statuses, etc. :)

**Comment by ibnux** [comment]
I see that he already implemented database using postgree or maybe MySQL
We can get message in there

But yeah it will be good if there a config in .env maybe like `url_message_received_callback=` if it not empty, then it post message to that url

**Comment by Diden05** [comment]
> I see that he already implemented database using postgree or maybe MySQL We can get message in there
> But yeah it will be good if there a config in .env maybe like `url_message_received_callback=` if it not empty, than it post message to that url

Only sqlite so far :)
MySQL would be magical, but whatsmeow doesn't plan mysql support, postgresql is prioritized there
PS: whatsmeow doesn't seem to store messages in the database

**Comment by dimaskiddo** [comment]
@Diden05 for PostgreSQL support you just need add postgresql connector in pkg/whatsapp/drivers.go and change the environment vars for Type and URI

**Comment by Diden05** [comment]
@dimaskiddo I understand, I will do it in the future, you must first transfer the main application from MySQL to PostgreSQL :)
Thanx!

**Comment by Diden05** [comment]
I found the documentation https://pkg.go.dev/go.mau.fi/whatsmeow#Client.AddEventHandler but somehow I have no idea how to do it in our case

**Comment by dimaskiddo** [comment]
Thank's @Diden05 for the information, i also already explore it, with implement it as webhook like pak @ibnux said soon

**Comment by sicenul** [comment]
Can't wait to see this in action... Any estimation on when to implement this pak @dimaskiddo ? :blush: :pray:

**Comment by ibnux** [comment]
For now I use android apps, whatsauto In playstore

Because WhatsApp still need dedicated phone that's why it handle received message in phone

**Comment by hrizal** [comment]
waiting for webhook implementation, this is going to be really cool 🙏

**Comment by dima-kov** [comment]
hey, really need this feature man. Do you have any estimations?

**Comment by dimaskiddo** [comment]
Hi all sorry for long long not updating here, i don't have any estimation yet because i just need to think more to not having deadlock since this API can do multiple account login
